internal/provider/kubernetes/gateway.go

```diff
@@ -30,19 +30,19 @@ type gatewayReconciler struct {
 	log             logr.Logger
 
 	initializeOnce sync.Once
-	resourceTable  *ResourceTable
+	resources      *message.ProviderResources
 }
 
 // newGatewayController creates a gateway controller. The controller will watch for
 // Gateway objects across all namespaces and reconcile those that match the configured
 // gatewayclass controller name.
-func newGatewayController(mgr manager.Manager, cfg *config.Server, resourceTable *ResourceTable) error {
-	resourceTable.Initialized.Add(1)
+func newGatewayController(name string, mgr manager.Manager, logger logr.Logger, resources *message.ProviderResources) error {
```

Comment:
Instead of passing in `logger logr.Logger`, consider using a context and then calling `logr.FromContext(ctx)`, [xref](https://github.com/envoyproxy/gateway/blob/main/internal/infrastructure/translator.go#L14-L24).

Reply:
I always prefer using function args to pass data around over context unless I absolutely need to ( and when the data is tied to a context rather than the package).
I can change the function signature and use a Config object instead to reduce the number of params passed to it